crates/bevy_ecs/src/component.rs

```diff
@@ -392,6 +420,35 @@ pub trait Component: Send + Sync + 'static {
     }
 }
 
+mod private {
+    pub trait Seal {}
+}
+
+/// The mutability option for a [`Component`]. This can either be:
```

Comment:
This should probably be where the main explanation lives. Move the docs here, and link from Component, then link this trait from the other marker types in this module.

Reply:
Agreed! I've moved the documentation around.